### PR TITLE
Fix the green line ignoring 'mark as read'

### PR DESCRIPTION
**Pull request by tienifr**
### Fixed Issues

$ https://github.com/Expensify/App/issues/32509
PROPOSAL: https://github.com/Expensify/App/issues/32509#issuecomment-1842591272

### Tests

1. Right click on any conversation from LHN> Mark as unread
2. Right click on the same conversation from LHN> Mark as read
3. Open the conversation
4. Verify that green line is not appear

- [x] Verify that no errors appear in the JS console

### Offline tests

Same as above

### QA Steps

1. Right click on any conversation from LHN> Mark as unread
2. Right click on the same conversation from LHN> Mark as read
3. Open the conversation
4. Verify that green line is not appear

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
        - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/113963320/d1edf20f-09e4-44c7-9eef-d3645756973e

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/113963320/c6f7e7f2-e7db-4e71-8f0c-8a1bef6509cb

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/113963320/db5455b2-80e4-4acc-8045-957afa850af1

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/113963320/c6d1d891-d929-4d3c-a946-36d1e5fd77d1

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/113963320/5b608743-844d-492b-9b68-6a40bf21d4f9

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/113963320/34dc61bb-25a3-4f1f-a1a1-8fc00687f3aa

</details>

### CODE REVIEW

File: src/pages/home/report/ReportActionsList.js [hunk 1]
```diff
@@ -228,12 +228,21 @@ function ReportActionsList({
             unreadActionSubscription.current = null;
         }
 
-        // Listen to specific reportID for unread event and set the marker to new message
-        unreadActionSubscription.current = DeviceEventEmitter.addListener(`unreadAction_${report.reportID}`, (newLastReadTime) => {
+        const resetUnreadMarker = (newLastReadTime) => {
             cacheUnreadMarkers.delete(report.reportID);
             lastReadTimeRef.current = newLastReadTime;
             setCurrentUnreadMarker(null);
             setMessageManuallyMarkedUnread(new Date().getTime());
```
Comment (cubuspl42) [inline src/pages/home/report/ReportActionsList.js:235]: Is this still true?

In one of our cases, we mark the action as "read",  but in the code, we execute the expression `setMessageManuallyMarkedUnread(...)`.

File: src/pages/home/report/ReportActionsList.js [hunk 2]
```diff
@@ -228,12 +228,21 @@ function ReportActionsList({
...
+        };
+
+        // Listen to specific reportID for unread event and set the marker to new message
+        unreadActionSubscription.current = DeviceEventEmitter.addListener(`unreadAction_${report.reportID}`, (newLastReadTime) => {
+            resetUnreadMarker(newLastReadTime);
+        });
+
+        // Listen to specific reportID for read newest action event and reset the marker
+        unreadActionSubscription.current = DeviceEventEmitter.addListener(`readNewestAction_${report.reportID}`, (newLastReadTime) => {
```
Comment (cubuspl42) [inline src/pages/home/report/ReportActionsList.js:244]: Is this intended? We store two different listener handles in a single ref.

**Comment by cubuspl42** [comment]
Please don't include bits like "Fix... {issue ID}" in the PR title / commit messages, as it triggers GitHub auto-closing behavior. Please force-push the branch with rewritten commit messages.

File: src/pages/home/report/ReportActionsList.js [hunk 3]
```diff
@@ -227,14 +229,28 @@ function ReportActionsList({
...
+        unreadActionSubscription.current = DeviceEventEmitter.addListener(`unreadAction_${report.reportID}`, (newLastReadTime) => {
+            resetUnreadMarker(newLastReadTime);
             setMessageManuallyMarkedUnread(new Date().getTime());
         });
+
+        // Listen to specific reportID for read newest action event and reset the marker
+        readNewestActionSubscription.current = DeviceEventEmitter.addListener(`readNewestAction_${report.reportID}`, (newLastReadTime) => {
+            resetUnreadMarker(newLastReadTime);
+            setMessageManuallyMarkedUnread(0);
+        });
```
Comment (cubuspl42) [inline src/pages/home/report/ReportActionsList.js:253]: - Make `ReportActionsList` a directory with `index.js` file
- Add a file `DeviceEventListenerRef.ts` in the new directory:

```ts
class DeviceEventListenerRef {
    // ...
    
    static use(
        type: string,
        listener: (data: unknown) => void
    ): DeviceEventListenerRef {
        // ...
    }

    add(): void {
        // ...
    }

    remove(): void {
        // ...
    }
}

```

File: src/pages/home/report/ReportActionsList/DeviceEventListenerRef.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import {DeviceEventEmitter, EmitterSubscription} from 'react-native';
+
+export default class DeviceEventListenerRef {
+    private typePrefix: string;
+
+    private subscription: EmitterSubscription | null = null;
+
+    constructor(typePrefix: string) {
+        this.typePrefix = typePrefix;
+    }
```
Comment (cubuspl42) [inline src/pages/home/report/ReportActionsList/DeviceEventListenerRef.ts:10]:
```ts
    constructor(private typePrefix: string) {
    }
```

File: src/pages/home/report/ReportActionsList/DeviceEventListenerRef.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+    static use(typePrefix: string): DeviceEventListenerRef {
+        return new DeviceEventListenerRef(typePrefix);
```
Comment (cubuspl42) [inline src/pages/home/report/ReportActionsList/DeviceEventListenerRef.ts:13]: I thought of an API that would directly reflect the previous usage. I meant a 100% direct refactoring without any change in the behavior.

We would use `useRef(null);` here and save it in the class private field. Then we'd use `DeviceEventListenerRef.use()` in the place we called `useRef(null);` before.

I'm **not** saying that what you did doesn't work, as it actually feels clearer.

We could go two ways:

- We could make the implementation directly reflect the old logic (my original intention)
- ...or, we could stay with your way, which is somewhat clearer, but:
  - we'd need to rename some things, as `DeviceEventListenerRef` and `use` names were directly related with the hook nature of my suggestion
  - we'd need to be 100% sure that this doesn't break anything, as the use-site of this logic is quite convoluted

**Review by jakub-trzebiatowski — CHANGES_REQUESTED** [review]

**Comment by cubuspl42** [comment]
About auto-closing, I meant this logic: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

It's not like I recommend naming PR / commits with the title of the _issue_ it solves, because it's also somewhat misleading... Words "Solved" /  "Fixed" often make sense in the sentence being part of the PR title or a commit message, just they shouldn't be connected with the _issue id_, especially with the leading hash. 

There's no need for an action right now, just in the future don't connect words "Fix/solve/close/... #issue-id" in the PR title or commits message. **Just** "Fixes..." is fine, just phrase something yourself, like "Fix the green line ignoring 'mark as read'"

**Comment by cubuspl42** [comment]
> I just wonder how we can use `useRef` in class? Please correct me if I misunderstand sth. I also re-test and confirm that it doesn't break any logics

The React hook system and code abstractions (like functions, classes...) don't care much about each other. The important bit is to stick to the naming conventions, i.e. name functions abstracting hooks `useFoo`, so on the use-site we follow the hook contracts. That's why I suggested naming the static class method `use`.

I originally meant something like this:

```ts
export default class DeviceEventListenerRef {
    constructor(
        private typePrefix: string,
        private subscription: MutableRefObject<EmitterSubscription | null>,
    ) {
    }

    static use(typePrefix: string): DeviceEventListenerRef {
        return new DeviceEventListenerRef(typePrefix, useRef(null));
    }

    add(
        typeSuffix: string,
        listener: (data: unknown) => void,
    ): void {
        this.subscription.current = DeviceEventEmitter.addListener(`${this.typePrefix}_${typeSuffix}`, listener);
    }

    remove(): void {
        if (!this.subscription.current) {
            return;
        }
        this.subscription.current.remove();
        this.subscription.current = null;
    }
}
```

This abstracts away the previously duplicated logic in a 1:1 manner.

I re-thought this, and **if** we were to go your way (depend on the `useEffect` cleanup function), the introduced class brings negative value. `EmitterSubscription` type returned by `DeviceEventEmitter.addListener` could be used directly.

I'm not sure why the `useEffect` cleanup function wasn't used here before, and I wouldn't be surprised if there's some gotcha hiding somewhere...

**Comment by cubuspl42** [comment]
> as the rule of react-hook we should not use the hook in the class

This is not true. You mean a "class component", not a "class", this is not the same thing. Sometimes abstracting some hook-related behavior with a class is practical, as long as all methods calling hooks use the `use*` prefix to clearly mark them as hooks. They also have to follow the hook contract themselves.

>  If we try it the following lint error will be thrown

This lint is confused, it mistakes a class with a class component.

> Please tell me what's your though?

As I said before, I'm fine with reorganizing the logic here as long as it's tested to work. So you can go ahead. You can revert the changes I suggested to introduce a class file. You can do that on top of the branch or rewrite the history with a force-push, whichever you prefer.

File: src/pages/home/report/ReportActionsList.js [hunk 6]
```diff
@@ -222,20 +224,30 @@ function ReportActionsList({
...
+
+        return () => {
+            // If the reportID changes, we reset the userActiveSince to null, we need to do it because
+            // this component doesn't unmount when the reportID changes
```
Comment (cubuspl42) [inline src/pages/home/report/ReportActionsList.js:247]: Outdated comment

File: src/pages/home/report/ReportActionsList.js [hunk 7]
```diff
@@ -145,6 +145,8 @@ function ReportActionsList({
     const opacity = useSharedValue(0);
     const userActiveSince = useRef(null);
     const unreadActionSubscription = useRef(null);
+    const readNewestActionSubscription = useRef(null);
```
Comment (cubuspl42) [inline src/pages/home/report/ReportActionsList.js:148]: Why are we using refs, if we don't access `.current` outside of a single anonymous function? Does this bring any value over using local `const`s?

```ts
const unreadActionSubscription = DeviceEventEmitter.addListener(...);
/// etc.
```

**Review by jakub-trzebiatowski — CHANGES_REQUESTED** [review]

File: src/pages/home/report/ReportActionsList.js [hunk 8]
```diff
@@ -222,20 +222,28 @@ function ReportActionsList({
     }, [report.lastReadTime, report.reportID]);
 
     useEffect(() => {
-        // If the reportID changes, we reset the userActiveSince to null, we need to do it because
-        // this component doesn't unmount when the reportID changes
```
Comment (cubuspl42) [inline src/pages/home/report/ReportActionsList.js:226]: Have you tested this case the comment mentions, i.e. when the "reportID changes"?

I believe the changes you suggested are clean and improve the code, but I just want to be sure we don't introduce a regression, even when the intentions are pure

File: src/pages/home/report/ReportActionsList.js [hunk 9]
```diff
@@ -222,20 +222,28 @@ function ReportActionsList({
...
-
-        // Listen to specific reportID for unread event and set the marker to new message
-        unreadActionSubscription.current = DeviceEventEmitter.addListener(`unreadAction_${report.reportID}`, (newLastReadTime) => {
+        const resetUnreadMarker = (newLastReadTime) => {
             cacheUnreadMarkers.delete(report.reportID);
             lastReadTimeRef.current = newLastReadTime;
             setCurrentUnreadMarker(null);
+        };
+
+        // Listen to specific reportID for unread event and set the marker to new message
```
Comment (cubuspl42) [inline src/pages/home/report/ReportActionsList.js:231]: I just spent a few minutes trying to suggest language improvements in this comment (and the similar one), but this comment just doesn't bring any value.

You can remove this comment and the other one.

File: src/pages/home/report/ReportActionsList.js [hunk 10]
```diff
@@ -222,20 +222,28 @@ function ReportActionsList({
...
+        const unreadActionSubscription = DeviceEventEmitter.addListener(`unreadAction_${report.reportID}`, (newLastReadTime) => {
+            resetUnreadMarker(newLastReadTime);
             setMessageManuallyMarkedUnread(new Date().getTime());
         });
+
+        // Listen to specific reportID for read newest action event and reset the marker
```
Comment (cubuspl42) [inline src/pages/home/report/ReportActionsList.js:237]: ☝️

**Comment by cubuspl42** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos

<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/2590174/2f64a16d-0bea-4940-9dc1-d217b1020a23

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/2590174/287da32f-8e73-4dcd-a57b-5e39ac297002

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/2590174/ce11e1a9-70d2-4039-87af-5ee04442bb63

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/2590174/8506b823-86a2-490f-9bd6-5192c3f0c970

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/2590174/442d4bf4-e49e-43f9-a5d1-13119f093c87

</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Review by jakub-trzebiatowski — APPROVED** [review]

**Comment by johnmlee101** [comment]
@cubuspl42 it looks like MacOS: Desktop is missing?

**Comment by johnmlee101** [comment]
Also @tienifr are you unable to upload screenshots/videos from Android?

**Comment by cubuspl42** [comment]
@johnmlee101 

> @cubuspl42 it looks like MacOS: Desktop is missing?

Sorry; fixed

**Review by johnmlee101 — APPROVED** [review]

**Pull request merged**